silx/gui/plot/items/histogram.py

```diff
@@ -297,6 +301,20 @@ def setData(self, histogram, edges, align='center', copy=True):
 
         self._updated(ItemChangedType.DATA)
 
+    def setBaseline(self, baseline):
+        if isinstance(baseline, numpy.ndarray):
+            self._baseline = numpy.empty(baseline.shape[0] * 2)
+            for i_value, value in enumerate(baseline):
+                self._baseline[i_value*2:i_value*2+2] = value
+        else:
+            self._baseline = baseline
+
+    def getBaseline(self, copy=True):
```

Comment:
The docstring is missing.

Defining a `BaselineMixIn` class to share the baseline implementation would allow to share the implementation, provide a fixed way of checking if the item has a baseline with `isinstance` and be consistent with the other items.

Also it doesn't cost much and is safer to always return a new array: `return numpy.array(self._baseline, copy=copy)